subprojects/docs/src/docs/userguide/api/kotlin_dsl.adoc

```diff
@@ -702,6 +702,38 @@ tasks.create("myTask") {
 // === `Property`, `Provider` and `NamedDomainObjectProvider`
 
 
+[[kotdsl:assignment]]
+== Kotlin lazy property assignment
+
+Gradle's Kotlin DSL supports lazy property assignment using the `=` operator as an experimental feature that can be enabled by adding `systemProp.org.gradle.unsafe.kotlin.assignment=true` to the `gradle.properties` file.
+Lazy property assignment reduces the verbosity for Kotlin DSL when <<lazy_configuration#lazy_properties,lazy properties>> are used.
+It works for properties that are publicly seen as `final` (without a setter) and have type `Property` or `ConfigurableFileCollection`.
+Since properties have to be `final`, our general recommendation is to not implement custom setters for properties with lazy types and if possible implement such properties via an abstract getter.
```

Comment:
💄 
```suggestion
Since properties have to be `final`, our general recommendation is not to implement custom setters for properties with lazy types and, if possible, implement such properties via an abstract getter.
```